### PR TITLE
feat: set up initial POC for Web UI

### PR DESCRIPTION
**Pull request by goetzrobin**
**Please describe the purpose of this pull request.**
Is it to add a new feature? Is it to fix a bug?

A new feature adding a Web-Based UI that allows users to interact with MemGPT outside of the CLI.

**How to test**
How can we test your PR during review? What commands should we run? What outcomes should we expect?

Please give it a try and see if you can get it up and running with the following steps:
1. run `memgpt configure`
2. run `memgpt server`
3. go to `http://127.0.0.1:8283` and see if you can create an agent (prefilled with data that works, need to add endpoints to fetch valid options) and chat with them

**Have you tested this PR?**
Have you tested the latest commit on the PR? If so please provide outputs from your tests.

I have tested it and was able to create an Agent and Chat with the agent

**Related issues or PRs**
Please link any related GitHub [issues](https://github.com/cpacker/MemGPT/issues) or [PRs](https://github.com/cpacker/MemGPT/pulls).

**Is your PR over 500 lines of code?**
If so, please break up your PR into multiple smaller PRs so that we can review them quickly, or provide justification for its length.

Justification is that initial setup was necessary to get up and running. Smaller PRs going forward!

**Additional context**
Add any other context or screenshots about the PR here.

### CODE REVIEW

File: memgpt/cli/cli.py [hunk 1]
```diff
@@ -106,6 +106,7 @@ def run(
     model_endpoint_type: str = typer.Option(None, help="Specify the LLM model endpoint type"),
     context_window: int = typer.Option(None, help="The context window of the LLM you are using (e.g. 8k for most Mistral 7B variants)"),
     # other
+    server: bool = typer.Option(False, "--server", help="Start MemGPT as a Web Server"),
```
Comment (cpacker) [inline memgpt/cli/cli.py:109]: Do we need this? I haven't look at the chain of execution but I started with `memgpt server` and it worked - does this code get touched?

File: memgpt/server/server.py [hunk 2]
```diff
@@ -443,7 +443,7 @@ def create_agent(
     def list_agents(self, user_id: str) -> dict:
         """List all available agents to a user"""
         agents_list = utils.list_agent_config_files()
-        return {"num_agents": len(agents_list), "agent_names": agents_list}
+        return {"num_agents": len(agents_list), "agents": [AgentConfig.load(f).__dict__ for f in agents_list]}
```
Comment (cpacker) [inline memgpt/server/server.py:446]: Nice catch

**Comment by cpacker** [comment]
image

LGTM for an init off the feature branch pending the one comment on the cli file, will merge into feat branch now so people can PR off the main repo for changes.

**Comment by goetzrobin** [comment]
@cpacker closing this for https://github.com/cpacker/MemGPT/pull/625 which has a cleaner commit history and removed the --server flag!

**Pull request closed without merging**